ui/app/components/ui/message-circle-icon/message-circle-icon.component.js

```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import CircleIcon from '../circle-icon'
+
+import danger from '../../../../../app/images/icons/red-triangle-exclaim.svg'
+import warning from '../../../../../app/images/icons/yellow-bell.svg'
+
+const typeConfig = {
+  danger: {
+    circleClass: 'message-circle-icon--danger',
+    iconSource: danger,
+  },
+  warning: {
+    circleClass: 'message-circle-icon--warning',
+    iconSource: warning,
+  },
+}
+
+export default class MessageCircleIcon extends Component {
+  static propTypes = {
+    type: PropTypes.string.isRequired,
```

Comment:
Nit: It might be helpful to be more specific here, so we're alerted if this component is used without a valid type.
```suggestion
    type: PropTypes.oneOf(Object.keys(typeConfig)).isRequired,
```

Reply:
Done in 47c627a7a